app/controllers/submit.js

```diff
@@ -1315,6 +1388,25 @@ export default Controller.extend(Analytics, BasicsValidations, COIValidations, N
             this.set('selectedProvider', this.get('currentProvider'));
             this.set('providerChanged', false);
         },
+        saveAuthorAssertions() {
+            this.get('metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: `${this.get('editMode') ? 'Edit' : 'Submit'} - Save and Continue Author Assertions Section`,
+                });
+            const model = this.get('model');
+            model.set('hasDataLinks', this.get('hasDataLinks'));
+            model.set('dataLinks', this.get('dataLinks'));
+            model.set('whyNoData', this.get('whyNoData'));
```

Comment:
Should `dataLinks` and `whyNoData` be set on the model all the time? Or should these be set for `available` and `no` respectively?

Reply:
We have other mechanisms in place to make sure `dataLinks` is only set on the compute when `hasDataLinks === 'available'` and `whyNoData` is defined when `hasDataLinks==='no'`, so this should be fine.